app/src/common/components/DeleteDialog.tsx

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+
+import {
+  Button,
+  Dialog,
+  DialogProps as MuiDialogProps,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  DialogActions,
+  makeStyles,
+  Typography,
+  CircularProgress,
+} from "@material-ui/core";
+import clsx from "clsx";
+import { useTranslation } from "react-i18next";
+
+import Alert from "common/components/Alert";
+
+type DeleteDialogProps = MuiDialogProps & {
+  title: string;
+  loading: boolean;
+  onDelete: React.MouseEventHandler<HTMLButtonElement> | undefined;
+  alert?: string;
+  onAlertClose: (
+    event?: React.SyntheticEvent<Element, Event> | undefined,
+    reason?: string | undefined
+  ) => void;
+};
```

Comment:
You shouldn't be needing this. Alert should be handled by the parent component, the one making the delete request